tket2-py/tket2/tket2/_circuit.pyi

```diff
@@ -0,0 +1,106 @@
+from enum import Enum
+from typing import Any, Callable
+from pytket._tket.circuit import Circuit
+
+class Tk2Circuit:
+    """Rust representation of a TKET2 circuit."""
+
+    def __init__(self, circ: Circuit) -> None:
+        """Create a Tk2Circuit from a pytket Circuit."""
+    def __hash__(self) -> int:
+        """Compute the circuit hash by traversal."""
+    def circuit_cost(self, cost_fn: Callable[[Tk2Op], Any]) -> int:
+        """Compute the cost of the circuit. Return value must implement __add__."""
+    def node_op(self, node: Node) -> CustomOp:
+        """If the node corresponds to a custom op, return it. Otherwise, raise an error."""
+    def to_tket1(self) -> Circuit:
+        """Convert to pytket Circuit."""
+    def apply_rewrite(self, rw) -> None:
+        """Apply a rewrite to the circuit."""
+    def node_inputs(self, node: Node) -> list[Wire]:
+        """The incoming wires to a node."""
+    def node_outputs(self, node: Node) -> list[Wire]:
+        """The outgoing wires from a node."""
+    def input_node(self) -> Node:
+        """The input node of the circuit."""
+    def output_node(self) -> Node:
+        """The output node of the circuit."""
+
+class Tk2Op(Enum):
+    """A Tket2 built-in operation."""
+
+    H = 1
+    CX = 2
```

Comment:
Add all the others? At least the ones that appear in `build.py`.
```suggestion
    CX = auto()
    T = auto()
    S = auto()
    X = auto()
    Y = auto()
    Z = auto()
    Tdg = auto()
    Sdg = auto()
    ZZMax = auto()
    Measure = auto()
    RzF64 = auto()
    RxF64 = auto()
    PhasedX = auto()
    ZZPhase = auto()
    AngleAdd = auto()
    CZ = auto()
    TK1 = auto()
    QAlloc = auto()
    QFree = auto()
    Reset = auto()
```
It's annoying that the values are required. They do not exist on the rust side.